NDTensors/src/BlockSparseArrays/src/sparsearray.jl

```diff
@@ -0,0 +1,31 @@
+struct SparseArray{T,N} <: AbstractArray{T,N}
+  data::Dictionary{CartesianIndex{N},T}
+  dims::NTuple{N,Int64}
```

Comment:
Reminder to add a `zero` field, i.e.:
```suggestion
  data::Dictionary{CartesianIndex{N},T}
  dims::NTuple{N,Int64}
  zero::Zero
```
This would be a custom `zero` function for getting zero elements that aren't just determined from the element type as `zero(T)`, for example when the elements are blocks, in which case the `zero` function would need to know about the block sizes. It should be a function of `T` and the `CartesianIndex` `I`, and could default to `zero(T)`.

Reply:
Completed.